rfc/rfc-92/rfc-92.md

```diff
@@ -0,0 +1,200 @@
+
+  
+## **Implementation**
+
+The main implementation step here is to create abstraction called TableFormatPlugin which handles table format operations such as 
+
+
+1. Committing writes 
+2. Update and read table format specific Metadata operations 
+3. Timeline 
+4. Conflict Resolution
+5. Lock Provider
+6. Rollbacks
+
+
+The Hudi platform is responsible for managing the data path and can be configured with the table format plugin by default using Hudi native table format. Other table formats will have their own implementation of this abstraction.  
+
+### Commit Protocol:
+
+Hudi uses the creation of a .commit file to advertise the completion of a commit on the table. With the TableFormatPlugin, this will be a two step process:
+
+1.  Creation of .commit file in .hoodie timeline 
+2. Store the commit completion time in the table format's commit metadata.
+
+  
+With this, the write is only completed when both the above steps are completed. The plugin provided timeline needs to fence the timeline ensuring the definition of complete stays consistent.  This ensures the snapshot isolation is maintained.
+
+### Metadata:
+
+If a different table format is configured, Hudi's metadata operations are replaced with the table format's metadata. This is done by adding a new adapter implementation of HoodieTableMetadata.
+
+
+### Timeline: 
+
+The timeline implementation ensures the external table format's (table format that is plugged-in) metadata state (e:g commit status) is the source of truth for all operations. 
+
+### Others:
+
+The plugin implementation brings in the external table format's conflict resolution strategy and locking mechanisms. 
+
+### Layout:
+
+The metadata corresponding to the table format (iceberg, ..) will be stored under .hoodie/ folder.
+
+### Codebase:
+
+The pluggable interface and Hudi's native format implementation will reside in hudi codebase while support for other systems will be done in X-table.
+
+
+The lakehouse platform interacts with this plugin and the table can be queried using the table format corresponding to the plugin. Here is the interface definition:
+
+```
+/**
+ * External Table Format needs to implement this class.
+ * Contract: 
+ *    This implementation will get the commit and completeXXX calls after the Hudi platform commit's internally. 
+ *    An operation must be defined as complete only when the plugin's implementation successfully handles the corresponding callback.
+ *    This implementation is responsible for providing the source of truth timeline based on what operations completed successfully.
+ */
+public interface PluggableTableFormat implements Serializable {
```

Comment:
just `HoodieTableFormat`?

Reply:
will we move all current impls to a `NativeTableFormat` that implements `HoodieTableFormat`?

Reply:
Yes. Updated.